crates/data_components/src/flight.rs

```diff
@@ -538,32 +538,39 @@ fn query_to_stream(
 fn to_execution_error(e: Error) -> DataFusionError {
     match e {
         Error::Flight { source } => match source {
-            flight_client::Error::UnableToQuery { source } => {
-                DataFusionError::Execution(format!("{source}"))
+            flight_client::Error::UnableToQuery {
+                source: source_error,
+            } => DataFusionError::Execution(format!("{source_error}")),
+            flight_client::Error::ConnectionReset { source: _ } => {
```

Comment:
```suggestion
            flight_client::Error::UnableToQuery {
                source,
            } => DataFusionError::Execution(format!("{source}")),
            flight_client::Error::ConnectionReset { .. } => {
```

nit: This can be more concise - there is no reason to specify `source: _` when we can just drop the struct. There is no reason to override `source: source_error` when we can overload `source` for the scope of that match.